test/org_parser/core_test.cljc

```diff
@@ -8,3 +8,31 @@
   (testing "minimal"
     (let [minimal (slurp "test/org_parser/fixtures/minimal.org")]
       (is (= minimal (-> minimal core/read-str core/write-str))))))
+
+
+(deftest headline-data
+  (is (= (core/read-str "* foo bar")
+         {:headlines [{:headline {:level 1,
+                                  :title [[:text-normal "foo bar"]],
+                                  :planning [],
+                                  :keyword nil,
+                                  :priority nil,
+                                  :commented? false
+                                  :tags []}}]}))
+  (is (= (core/read-str "* TODO COMMENT foo bar")
```

Comment:
Does it also work for just `* COMMENT foo bar`? A test for that would be good, too.

Reply:
It does not, but that seems to be an issue with the parser AFAICT. It reads the `COMMENT` as a keyword like `TODO` when one is not present:

```clojure
  (require '[org-parser.parser :refer [parse]])

  (parse "* COMMENT foo bar")
  ;; => [:S
  ;;     [:headline
  ;;      [:stars "*"]
  ;;      [:keyword "COMMENT"]
  ;;      [:text [:text-normal "foo bar"]]]]
```

Reply:
Only if EBNF cannot do this everything-except thing, this must be fixed by the transformer that you're working on. But it may work with negative look-ahead or so.

Reply:
Yeah makes sense. I have `NEXT`, `WAITING` etc. defined in my own personal setup 😄 I'd want it to support that.

I can try to make that change in the grammar. Perhaps I could address that as part of a follow up PR? I'd have to brush up on my EBNF-fu.

In the meantime, I could add a special case to the transform reducer to look for either a keyword = "COMMENT" or an `:inline-comment` node.

Reply:
It would be great if you can brush up your EBNF-fu :D you're welcome to include it in this PR. Maybe it's easy, then we don't need the transform reducer hack ;)